examples/How_to_use_moderation.ipynb

```diff
@@ -0,0 +1,455 @@
+   "outputs": [],
+   "source": [
+    "from openai import OpenAI\n",
+    "client = OpenAI()\n",
+    "\n",
+    "GPT_MODEL = 'gpt-3.5-turbo'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 1. Input moderation\n",
+    "Input Moderation focuses on preventing harmful or inappropriate content from reaching the LLM, with common applications including:\n",
+    "- **Content Filtering:** Prevent the spread of harmful content such as hate speech, harassment, explicit material, and misinformation on social media, forums, and content creation platforms.\n",
+    "- **Community Standards Enforcement:** Ensure that user interactions, such as comments, forum posts, and chat messages, adhere to the community guidelines and standards of online platforms, including educational environments, gaming communities, or dating apps.\n",
+    "- **Spam and Fraud Prevention:** Filter out spam, fraudulent content, and misleading information in online forums, comment sections, e-commerce platforms, and customer reviews.\n",
+    "\n",
+    "These measures act as preventive controls, operating before or alongside the LLM to alter your application's behavior if specific criteria are met."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Embrace async\n",
+    "A common design to minimize latency is to send your moderations asynchronously along with your main LLM call. If your moderation gets triggered you send back a placeholder response, otherwise send back the LLM response. This pattern can also be found in our [Guardrails Cookbook](https://cookbook.openai.com/examples/how_to_use_guardrails).\n",
+    "\n",
+    "We'll use this approach, creating an execute_chat_with_moderation function that will run our LLM's get_chat_response and the check_expression moderation function in parallel, and return the LLM response only if the moderation returns False (not triggered).\n",
+    "\n",
+    "#### Workflow:\n",
+    "We will create a workflow that incorporates the Moderation API to check user input for any potentially unsafe content before it is sent to the Language Model (LLM). This ensures that only appropriate content is processed further in the application.\n",
+    "\n",
+    "1. Receive an input from the user\n",
+    "2. Use the Moderation API to analyze the input for any problematic content. \n",
+    "3. **Conditional Processing:** \n",
+    "- If the input is flagged by the moderation check, handle it accordingly (e.g., reject the input, ask the user to rephrase, etc.).\n",
+    "- If the input is not flagged, pass it to the LLM for further processing.\n",
+    "\n",
+    "We will demonstrate this workflow with two example prompts:\n",
+    "\n",
+    "- **Bad Prompt:** An example input that should trigger the moderation categories and be blocked from reaching the LLM.\n",
+    "- **Good Prompt:** An example input that should pass the moderation check and be sent to the LLM."
```

Comment:
nit: don't think you need to explain these two

Reply:
removed